### ISSUE TITLE
NNPA Performance regression in several models after llvm uplift

### ISSUE THREAD

**Issue by cjvolzka** [onnx/onnx-mlir#2634]
In the following models, roberta-base-11 t5-decoder-with-lm-head-12 resnet18-v2-7 resnet34-v2-7 resnet50-v2-7 resnet101-v2-7 resnet152-v2-7, there's a notable performance regression on NNPA between commit dce84b5 (good) and 96a2558 (regression).  See picture below for specific runtimes in ms. 

* All models compiled with `--O3 --EmitLib --mtriple=s390x-ibm-loz --mcpu=z16 --maccel=NNPA  --onnx-op-stats TXT`
* runtimes in ms and are an average of 10 inferences after a single warmup inference
* used the dataset inputs from the onnx model zoo for each model
* Other models may be impacted these are just the ones I confirmed are affected by this commit specifically
![image](https://github.com/onnx/onnx-mlir/assets/42243335/c520de97-118f-4f16-81e9-4d41db39e0d0)

**Comment by AlexandreEichenberger** [comment]
I believe this is fixed by merged #2652

**Comment by cjvolzka** [comment]
Confirmed. Runtime for the models above is back to previous norms:
Results from testing commit 826f9af7
```
roberta-base-11.onnx averaging 7.23199 ms 
t5-decoder-with-lm-head-12.onnx averaging 10.2682 ms 
resnet18-v2-7.onnx averaging 13.3078 ms 
resnet34-v2-7.onnx averaging 16.9058 ms 
resnet50-v2-7.onnx averaging 30.2422 ms 
resnet101-v2-7.onnx averaging 44.4475 ms 
resnet152-v2-7.onnx averaging 59.0743 ms 
```
